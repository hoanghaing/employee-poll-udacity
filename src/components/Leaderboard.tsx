import React, { useEffect, useState } from 'react';
import { Navigate } from 'react-router-dom';

const Leaderboard = () => {
  return (
    <>
      <p>Leaderboard panel</p>
    </>
  );
}
export default Leaderboard