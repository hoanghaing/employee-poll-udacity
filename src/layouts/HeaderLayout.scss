.tabs {
  display: inline-block;
  border-bottom: 2px solid rgb(168, 168, 168);
  width: -moz-available;
  margin-left: 2rem;
  margin-right: 2rem;
  .tab {
    margin-top: 0.5rem;
  }
}
.user {
  display: flex;
  margin: 0.5rem auto;
  float: right;
  align-items: center;
  .avt {
    width: 2rem;
    margin-right: 0.5rem;
  }
  .name {
    font-weight: 600;
  }
}
.log-out {
  display: inline;
  margin-left: 1rem;
  margin-top: 0.65rem;
  &:hover {
    cursor: pointer;
    color: rgb(112, 112, 112);
    font-weight: 500;
  }
  float: right;
}