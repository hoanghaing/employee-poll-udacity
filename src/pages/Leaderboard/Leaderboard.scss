.ranking-table {
  width: 60rem;
  border: 1px solid #c7c7c7;
  border-radius: 4px;
  margin: 1rem;
  margin-left: auto;
  margin-right: auto;
}