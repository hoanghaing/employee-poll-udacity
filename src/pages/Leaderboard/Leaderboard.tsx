import React, { useEffect, useState } from 'react';
import './Leaderboard.scss';
import * as api from '@/api/index';

const Leaderboard = () => {
  const [rankings, setRanking] = useState([]);
  const itemsComponent: JSX.Element[] = [];
  useEffect(() => {
    const result = api._getRanking();
    setRanking(result);
  }, []);
  for (const user of rankings) {
    itemsComponent.push(
      <tr>
        <td>
          <div className="flex items-center gap-3">
            <div className="avatar">
              <div className="mask mask-squircle w-12 h-12">
                <img src={`../src/${user.avatarURL}`} alt="Avatar Tailwind CSS Component" />
              </div>
            </div>
            <div>
              <div className="font-bold">{`${user.name}`}</div>
              <div className="text-sm opacity-50">{`${user.id}`}</div>
            </div>
          </div>
        </td>
        <td>
          {user.answered}
        </td>
        <td>
          {user.created}
        </td>
      </tr>
    );
  }
  return (<div className="ranking-table overflow-x-auto">
    <table className="table">
      <thead>
        <tr>
          <th>Users</th>
          <th>Answered</th>
          <th>Created</th>
        </tr>
      </thead>
      <tbody>
        {itemsComponent}
      </tbody>
    </table>
  </div>)
}
export default Leaderboard