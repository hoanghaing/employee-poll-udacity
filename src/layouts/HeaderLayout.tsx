import React from 'react';
import './HeaderLayout.scss';
import { useNavigate } from "react-router-dom";
const HeaderLayout = ({ children }) => {
  const navigate = useNavigate();
  const onClickLogout = () => {
    localStorage.removeItem("authenticated");
    return navigate("/login");
  }
  return (
    <div>
      <header>
        <div role="tablist" className="tabs">
          <a role="tab" className="tab" href='/home'>Home</a>
          <a role="tab" className="tab" href='/leaderboard'>Leaderboard</a>
          <a role="tab" className="tab" href='/new'>New</a>
          <div className='log-out' onClick={onClickLogout}>
            Logout
          </div>
          <div className='user'>
            <img className='avt' src='./src/assets/avatar/mtsamis.jpg'/>
            <span className='name'>Ahihi</span>
          </div>

        </div>
      </header>
      {children}
    </div>
  );
};

export default HeaderLayout;