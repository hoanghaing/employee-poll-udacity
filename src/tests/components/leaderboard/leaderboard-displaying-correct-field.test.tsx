import { expect, describe, it, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import HeaderLayout from '@/layouts/HeaderLayout';
import Leaderboard from '@/pages/Leaderboard/Leaderboard';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

// Create a mock Redux store
const mockStore = configureStore([]);
const store = mockStore({
  user: {
    id: '',
    name: '',
    avatarURL: ''
  }
  // Mock your Redux state here if needed
});

test('Snapshot: renders leaderboard', async () => {
  const { debug, container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <HeaderLayout>
          <Leaderboard />
        </HeaderLayout>
      </BrowserRouter>
    </Provider>
  );
  // debug();
  expect(container.firstChild).toMatchSnapshot();
});
