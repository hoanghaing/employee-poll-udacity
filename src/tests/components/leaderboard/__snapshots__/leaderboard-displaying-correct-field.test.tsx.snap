// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders login form correctly 1`] = `
<div>
  <header>
    <div
      class="tabs"
      role="tablist"
    >
      <span
        class="tab"
        role="tab"
      >
        <a
          href="/home"
        >
          Home
        </a>
      </span>
      <span
        class="tab"
        role="tab"
      >
        <a
          href="/leaderboard"
        >
          Leaderboard
        </a>
      </span>
      <span
        class="tab"
        role="tab"
      >
        <a
          href="/new"
        >
          New
        </a>
      </span>
      <div
        class="log-out"
      >
        Logout
      </div>
      <div
        class="user"
      >
        <img
          class="avt"
          src="../src/"
        />
        <span
          class="name"
        />
      </div>
    </div>
  </header>
  <div
    class="ranking-table overflow-x-auto"
  >
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Users
          </th>
          <th>
            Answered
          </th>
          <th>
            Created
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div
              class="flex items-center gap-3"
            >
              <div
                class="avatar"
              >
                <div
                  class="mask mask-squircle w-12 h-12"
                >
                  <img
                    alt="Avatar Tailwind CSS Component"
                    src="../src//assets/avatar/sarahedo.jpg"
                  />
                </div>
              </div>
              <div>
                <div
                  class="font-bold"
                >
                  Sarah Edo
                </div>
                <div
                  class="text-sm opacity-50"
                >
                  sarahedo
                </div>
              </div>
            </div>
          </td>
          <td>
            4
          </td>
          <td>
            2
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="flex items-center gap-3"
            >
              <div
                class="avatar"
              >
                <div
                  class="mask mask-squircle w-12 h-12"
                >
                  <img
                    alt="Avatar Tailwind CSS Component"
                    src="../src//assets/avatar/mtsamis.jpg"
                  />
                </div>
              </div>
              <div>
                <div
                  class="font-bold"
                >
                  Mike Tsamis
                </div>
                <div
                  class="text-sm opacity-50"
                >
                  mtsamis
                </div>
              </div>
            </div>
          </td>
          <td>
            3
          </td>
          <td>
            2
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="flex items-center gap-3"
            >
              <div
                class="avatar"
              >
                <div
                  class="mask mask-squircle w-12 h-12"
                >
                  <img
                    alt="Avatar Tailwind CSS Component"
                    src="../src//assets/avatar/tylermcginnis.jpg"
                  />
                </div>
              </div>
              <div>
                <div
                  class="font-bold"
                >
                  Tyler McGinnis
                </div>
                <div
                  class="text-sm opacity-50"
                >
                  tylermcginnis
                </div>
              </div>
            </div>
          </td>
          <td>
            2
          </td>
          <td>
            2
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="flex items-center gap-3"
            >
              <div
                class="avatar"
              >
                <div
                  class="mask mask-squircle w-12 h-12"
                >
                  <img
                    alt="Avatar Tailwind CSS Component"
                    src="../src//assets/avatar/zoshikanlu.jpg"
                  />
                </div>
              </div>
              <div>
                <div
                  class="font-bold"
                >
                  Zenobia Oshikanlu
                </div>
                <div
                  class="text-sm opacity-50"
                >
                  zoshikanlu
                </div>
              </div>
            </div>
          </td>
          <td>
            1
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="flex items-center gap-3"
            >
              <div
                class="avatar"
              >
                <div
                  class="mask mask-squircle w-12 h-12"
                >
                  <img
                    alt="Avatar Tailwind CSS Component"
                    src="../src//assets/avatar/hainh.jpg"
                  />
                </div>
              </div>
              <div>
                <div
                  class="font-bold"
                >
                  Hai Nguyen Hoang
                </div>
                <div
                  class="text-sm opacity-50"
                >
                  hainh
                </div>
              </div>
            </div>
          </td>
          <td>
            0
          </td>
          <td>
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
